### ISSUE TITLE
Servers are occasionally shown "(Starting)" in UI while they're effectively started

### ISSUE THREAD

**Issue by adietish** [redhat-developer/vscode-adapters#324]
steps - this only happens occassionally, not every time:
1. ASSERT: have a server in vscode
1. EXEC: start it

Result:
![image](https://user-images.githubusercontent.com/25126/57537217-193f6200-7346-11e9-9ada-38b3bbf2a985.png)
Server is shown as (Started) even though it's effectively started already. The server is then shown as (Started) in UI and a user can't do anything but to kill it manually on the cmd-line to correct this: there's no "Restart", "Stop" in context menu.

**Comment by lstocchi** [comment]
I tried several times to replicate this error without any success. Do you have any hints to give me? Do you do any particular call/acton before starting a server or is it just random?

**Comment by adietish** [comment]
@lstocchi I had many many many servers defined (as you can see in the screenshot). Dont know if that matters, but worth a try?

**Comment by robstryker** [comment]
To replicate, make a program that uses port 8080 and just ties it up, ignoring all requests.  When wildfly starts, it won't be able to use 8080, and the poller will continue checking until 8080 responds with a 200 http code or something. Thus, the server adapter will be stuck in 'starting' mode.  

In the UI, it'd be nice to have a force-stop or terminate action.
